### ISSUE TITLE
Add OCI image annotations to Gradle/Maven plugin built images

### ISSUE THREAD

**Issue by candrews** [spring-projects/spring-boot#34047]
The images built by the Spring Boot Maven and Spring Boot Gradle plugins do not have [OCI image annotations](
https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys), such as `org.opencontainers.image.source`

These annotations are useful for people to manual use as well as for use by tools. For example, [Snyk uses them in its UI](https://snyk.io/blog/how-and-when-to-use-docker-labels-oci-container-annotations/) and [Renovate uses them to find release notes](https://github.com/renovatebot/renovate/blob/34.115.1/lib/modules/datasource/docker/readme.md).

I believe at least these labels should be added to the images:
* `org.opencontainers.image.source`
* `org.opencontainers.image.revision`
* `org.opencontainers.image.version`

Currently, it's possible to manually add these labels, enabled by the [image-labels buildpack](https://github.com/paketo-buildpacks/image-labels) which is included by default. For example, with gradle:
```groovy
tasks.named("bootBuildImage") {
	// See: https://github.com/paketo-buildpacks/image-labels
	// See: https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys)
	// GitLab: https://docs.gitlab.com/ee/ci/variables/predefined_variables.html
	// GitHub: https://docs.github.com/en/actions/learn-github-actions/variables
	// Azure DevOps Pipelines: https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables
	java.util.Optional.ofNullable(
		System.getenv("CI_PROJECT_URL")
		 ?: ( System.getenv("GITHUB_SERVER_URL") ? "${System.getenv("GITHUB_SERVER_URL")}/${System.getenv("GITHUB_REPOSITORY")}" : null)
		 ?: System.getenv("BUILD_REPOSITORY_URI")
	).ifPresent { environment["BP_OCI_SOURCE"] = it }
	java.util.Optional.ofNullable(
		System.getenv("CI_COMMIT_SHA")
		?: System.getenv("GITHUB_SHA")
		?: System.getenv("BUILD.SOURCEVERSION")
	).ifPresent { environment["BP_OCI_REVISION"] = it }
	environment["BP_OCI_VERSION"] = project.version
}
```
Relevant documentation: https://github.com/paketo-buildpacks/image-labels

I think this information should be set automatically, not requiring the user to do any such manual configuration.

**Comment by candrews** [comment]
> `Implementation-Version` entries in `META-INF/MANIFEST.MF`

I don't see `Implementation-Version` in the `META-INF/MANIFEST.MF` produced by my Spring Boot 3.0.2 project, so I'm thinking the first step is to get Spring Boot to populate `META-INF/MANIFEST.MF`  with that information. Would Spring Boot be open to doing that by default (if so I'll open an issue)?

**Comment by scottfrederick** [comment]
Maven will generate the `Implementation-*` entries with [some configuration](https://maven.apache.org/shared/maven-archiver/examples/manifest.html). I don't see a need for us to do anything there since it comes built-in with Maven. 

There is at least one [plugin available for Gradle](https://github.com/coditory/gradle-manifest-plugin) that provides similar functionality. I'm not sure we'd want to add this functionality to the Spring Boot Gradle plugin given that it's already possible without much effort. I'll tag this to see what other team members think.

**Comment by candrews** [comment]
I think telling users to have to setup a separate Gradle plugin or modify Maven configuration for something that would benefit them and therefore should be expected to Just Work ™️ goes against the design objectives of Spring Boot:

> takes an opinionated view of the Spring platform so that new and existing users can quickly get to the bits they need.

**Comment by wilkinsona** [comment]
We've configured Maven's jar plugin to add the `Implementation-*` entries [since 1.2.1](https://github.com/spring-projects/spring-boot/commit/91587d63a228feb5e9b29770778a36aa3c4edecb). We don't do the same for Gradle. Arguably that's a bug as we generally aim for consistency between the two build systems.

**Comment by scottfrederick** [comment]
Good catch, Andy. I've created #34059 to address the inconsistency between the Spring Boot Maven and Gradle plugins.
